### PR TITLE
Pass Variable Length Argument To Old Function Call

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:
- Variable length arguments not passed to `old_fn`

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smdebug/tensorflow/utils.py [hunk 1]
```diff
@@ -339,7 +339,7 @@ def get_layer_call_fn(layer: tf.keras.layers.Layer) -> Callable[[tf.Tensor], tf.
 
     def call(inputs, *args, **kwargs) -> tf.Tensor:
         layer_input = inputs
-        layer_output = old_call_fn(inputs)
+        layer_output = old_call_fn(inputs, *args, **kwargs)
```
Comment (vandanavk) [inline smdebug/tensorflow/utils.py:342]: was this causing a bug?

Comment (NihalHarish) [reply smdebug/tensorflow/utils.py:342]: Yes 
```
TypeError: call() missing 2 required positional arguments: 'training' and 'mask'
```

Comment (vandanavk) [reply smdebug/tensorflow/utils.py:342]: add a test for this case?

Comment (NihalHarish) [reply smdebug/tensorflow/utils.py:342]: Done.

File: tests/tensorflow2/test_model_subclassing.py [hunk 2]
```diff
@@ -26,7 +36,7 @@ def __init__(self):
     def first(self, x):
         with tf.name_scope("first"):
             tf.print("mymodel.first")
-            x = self.conv1(x)
+            x = self.conv1(x, input_one=1)
             # x = self.bn(x)
```
[the remark this answers is not in the document]
Comment (NihalHarish) [reply tests/tensorflow2/test_model_subclassing.py:40]: Done

File: tests/tensorflow2/test_model_subclassing.py [hunk 3]
```diff
@@ -13,6 +13,16 @@ def __init__(self):
         self.conv1 = Conv2D(
             32, 3, activation="relu", kernel_initializer=tf.keras.initializers.GlorotNormal(seed=12)
         )
+        self.original_call = self.conv1.call
+
+        def new_call(inputs, *args, **kwargs):
+            # Since we use layer wrapper we need to assert if these parameters
+            # are actually being passed into the original call fn
+            assert kwargs["input_one"] == 1
+            kwargs.pop("input_one")
```
[the remark this answers is not in the document]
Comment (NihalHarish) [reply tests/tensorflow2/test_model_subclassing.py:22]: Unexpected keyword argument error

**Review by vandanavk — APPROVED** [review]

**Pull request merged**